File: 5-longest-palindromic-substring/5-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        def helper(s, left, right):
            while left >= 0 and right < len(s) and s[left] == s[right]:
                left -= 1
                right += 1
            return s[left + 1 : right]
        res = ""
        for i in range(len(s)):
            tmp = helper(s, i, i)
            if len(tmp) > len(res):
                res = tmp
            tmp = helper(s, i, i + 1)
            if len(tmp) > len(res):
                res = tmp
        return res

File: 5-longest-palindromic-substring/5-longest-palindromic-substring.java
class Solution {
    String ans = "";
    
    public String longestPalindrome(String s) {
        for (int i = 0; i < s.length(); i++) {
            check(s, i, i);
            check(s, i, i + 1);
        }
        return ans;
    }
    
    public void check(String s, int left, int right) {
        while (left >= 0 && right < s.length()) {
            if (s.charAt(left) == s.charAt(right)) {
                left--;
                right++;
            } else {
                break;
            }
        }
        left++;
        right--;
        if (right - left + 1 > ans.length()) {
            ans = s.substring(left, right + 1);
        }
    }
}